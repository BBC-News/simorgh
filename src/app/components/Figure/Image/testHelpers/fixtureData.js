export const landscape = {
  alt: 'Student sitting an exam',
  src:
    'https://ichef.bbci.co.uk/news/1024/cpsprodpb/7098/production/_104842882_students.jpg',
  width: 1024,
  height: 576,
};

export const portrait = {
  alt:
    'Map of the UK displaying Syrian refugees and asylum seekers per 10000 population. Ranges from 0 to 17.',
  src:
    'https://ichef.bbci.co.uk/news/640/cpsprodpb/439A/production/_100960371_syrians_and_asylum_v2-nc.png',
  width: 640,
  height: 1138,
};

export const square = {
  alt: 'Tracks through the snow',
  src:
    'https://ichef.bbci.co.uk/news/1024/cpsprodpb/114FE/production/_104801907_79010.jpg',
  width: 1024,
  height: 1024,
};

export const custom = {
  alt: 'By Elisa Decker, from her series "Sidewalk"',
  src:
    'https://ichef.bbci.co.uk/news/445/cpsprodpb/164BB/production/_104032319_03270dcc-9dda-4bd4-96a0-db89f6b915ae.jpg',

  width: 445,
  height: 547,
};
