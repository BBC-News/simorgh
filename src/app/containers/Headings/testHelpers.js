const blocksSingleFragment = (text, attributes = []) => [
  {
    id: '7987213',
    type: 'text',
    model: {
      blocks: [
        {
          id: '6897239',
          type: 'paragraph',
          model: {
            text,
            blocks: [
              {
                id: '4209872',
                type: 'fragment',
                model: {
                  text,
                  attributes,
                },
              },
            ],
          },
        },
      ],
    },
  },
];

export default blocksSingleFragment;
