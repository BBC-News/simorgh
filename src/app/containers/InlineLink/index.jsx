import React, { useContext } from 'react';
import { Link } from 'react-router-dom';
import pathToRegexp from 'path-to-regexp';
import InlineLink from '@bbc/psammead-inline-link';
import VisuallyHiddenText from '@bbc/psammead-visually-hidden-text';
import { ServiceContext } from '../../contexts/ServiceContext';
import { RequestContext } from '../../contexts/RequestContext';
import Blocks from '../Blocks';
import fragment from '../Fragment';
import { inlineLinkModelPropTypes } from '../../models/propTypes/inlineLink';
import { articleRegexPath } from '../../routes/regex';

const InternalInlineLink = InlineLink.withComponent(Link);

const componentsToRender = { fragment };

const InlineLinkContainer = ({ locator, isExternal, blocks }) => {
  const { externalLinkText } = useContext(ServiceContext);
  const { platform } = useContext(RequestContext);

  const regexp = pathToRegexp(articleRegexPath, [], {
    start: false,
    end: false,
  });

  const result = regexp.exec(locator);
  // if URL matches a valid route, use a react-router link
  if (result) {
    // the path is the first item in the array
    let path = result[0];

    if (platform === 'amp') {
      path += '.amp';
    }

    return (
      <InternalInlineLink to={path}>
        <Blocks blocks={blocks} componentsToRender={componentsToRender} />
      </InternalInlineLink>
    );
  }

  // else return a normal hyperlink
  return (
    <InlineLink href={locator}>
      <Blocks blocks={blocks} componentsToRender={componentsToRender} />
      {isExternal ? (
        <VisuallyHiddenText>{externalLinkText}</VisuallyHiddenText>
      ) : null}
    </InlineLink>
  );
};

InlineLinkContainer.propTypes = inlineLinkModelPropTypes;

export default InlineLinkContainer;
