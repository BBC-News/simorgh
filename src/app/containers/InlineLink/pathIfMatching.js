import pathToRegexp from 'path-to-regexp';

const pathIfMatching = (regexString, url) => {
  const regexp = pathToRegexp(regexString, [], {
    start: false,
    end: false,
  });

  const regexResultArray = regexp.exec(url);

  if (!regexResultArray) {
    return null;
  }

  const path = regexResultArray[0];
  return path;
};

export default pathIfMatching;
