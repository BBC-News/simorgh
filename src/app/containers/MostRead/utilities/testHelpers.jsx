import React from 'react';
import MostReadContainer from '..';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContext } from '#contexts/ToggleContext';

export const setStalePromoTimestamp = mostReadData => {
  const oldTimestamp = 864691200; // 27/05/1997
  const updatedMostReadData = mostReadData;

  // set first promo to have an old timestamp
  updatedMostReadData.records[0].promo.timestamp = oldTimestamp;

  return updatedMostReadData;
};

export const setFreshPromoTimestamp = mostReadData => {
  const freshDate = new Date();
  const updatedMostReadData = mostReadData;

  // Updates first 10 promos to have a fresh date
  for (let i = 0; i < 10; i += 1) {
    updatedMostReadData.records[i].promo.timestamp = freshDate.getTime();
  }

  return updatedMostReadData;
};

export const setStaleLastRecordTimeStamp = mostReadData => {
  const updatedMostReadData = mostReadData;
  updatedMostReadData.lastRecordTimeStamp = '2019-11-06T16:28:00Z';

  return updatedMostReadData;
};

const getToggleState = enabled => ({
  local: { mostRead: { enabled } },
  test: { mostRead: { enabled } },
});

/* eslint-disable react/prop-types */
export const MostReadWithContext = ({
  isAmp = false,
  service,
  variant = null,
  mostReadToggle = false,
}) => (
  <ToggleContext.Provider
    value={{ toggleState: getToggleState(mostReadToggle) }}
  >
    <RequestContextProvider
      bbcOrigin={`http://localhost:7080/${service}`}
      isAmp={isAmp}
      pageType="frontPage"
      service={service}
      statusCode={200}
      pathname={`/${service}`}
      variant={variant}
    >
      <ServiceContextProvider service={service} variant={variant}>
        <MostReadContainer />
      </ServiceContextProvider>
    </RequestContextProvider>
  </ToggleContext.Provider>
);
/* eslint-enable react/prop-types */
