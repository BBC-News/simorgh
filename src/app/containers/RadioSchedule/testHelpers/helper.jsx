/* eslint-disable react/prop-types */
import { TEXT_VARIANTS } from '@bbc/psammead-storybook-helpers';
import { boolean } from '@storybook/addon-knobs';
import React from 'react';
import ProgramCard from '../ProgramCard';
import { ServiceContextProvider } from '#contexts/ServiceContext';

// Will remove and clean up in future PRs
export const stateTypes = ['live', 'onDemand', 'onDemand', 'next'];
export const uniqueStates = ['live', 'onDemand', 'next'];

export const renderProgramCard = ({
  state,
  script,
  service = 'news',
  duration = 'PT30M',
  durationLabel = 'Duration %duration%',
  startTime = 1566914061212,
  displaySummary = boolean('show summary', true),
  linkComponent = 'a',
  linkComponentAttr = 'href',
}) => {
  const { text, articlePath, longText, dir } = TEXT_VARIANTS[service];

  const props = {
    durationLabel: dir === 'rtl' ? 'المدة الزمنية %duration%' : durationLabel,
    linkComponent,
    linkComponentAttr,
  };

  const program = {
    state,
    link: articlePath,
    startTime,
    brandTitle: text,
    summary: displaySummary ? longText : null,
    duration,
  };

  return (
    <ServiceContextProvider
      script={script}
      service={service}
      locale="en-gb"
      dir={dir}
      timezone="Europe/London"
    >
      <ProgramCard program={program} {...props} />
    </ServiceContextProvider>
  );
};
