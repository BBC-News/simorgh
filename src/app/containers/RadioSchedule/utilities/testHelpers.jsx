import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import RadioScheduleContainer from '..';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContext } from '#contexts/ToggleContext';

const getToggleState = enabled => ({
  radioSchedule: { enabled },
});

/* eslint-disable react/prop-types */
const RadioSchedulesWithContext = ({
  service,
  radioScheduleToggle = false,
}) => (
  <ToggleContext.Provider
    value={{
      toggleState: getToggleState(radioScheduleToggle),
      toggleDispatch: jest.fn(),
    }}
  >
    <RequestContextProvider
      isAmp={false}
      pageType="frontPage"
      service={service}
      pathname={`/${service}`}
      timeOnServer={Date.now()}
    >
      <ServiceContextProvider service={service}>
        <BrowserRouter>
          <RadioScheduleContainer />
        </BrowserRouter>
      </ServiceContextProvider>
    </RequestContextProvider>
  </ToggleContext.Provider>
);
/* eslint-enable react/prop-types */

export default RadioSchedulesWithContext;
