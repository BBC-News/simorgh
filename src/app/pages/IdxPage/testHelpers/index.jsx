import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { ToggleContextProvider } from '#contexts/ToggleContext';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import IdxPage from '#pages/IdxPage';

const radioServiceOverride = 'dari';

// eslint-disable-next-line react/prop-types
const IdxPageWithContext = ({ service = 'persian', pageData }) => (
  <BrowserRouter>
    <ToggleContextProvider service={service} origin="https://www.test.bbc.com">
      <RequestContextProvider
        pageType="IDX"
        service={service}
        pathname="/pathname"
        data={{ status: 200 }}
        isAmp={false}
      >
        <ServiceContextProvider service={service}>
          <IdxPage
            pageData={pageData}
            radioScheduleEndpointOverride={`./data/${service}/bbc_${radioServiceOverride}_radio/schedule.json`}
          />
        </ServiceContextProvider>
      </RequestContextProvider>
    </ToggleContextProvider>
  </BrowserRouter>
);

export default IdxPageWithContext;
