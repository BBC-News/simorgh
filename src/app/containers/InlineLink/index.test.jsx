import React from 'react';
import { StaticRouter } from 'react-router-dom';
import { shouldMatchSnapshot } from '../../helpers/tests/testHelpers';
import InlineLink from './index';

describe('InlineLink', () => {
  describe('with link matching routes for SPA', () => {
    shouldMatchSnapshot(
      'should render correctly',
      /*
        for the value it would bring, it is much simpler to wrap a react-router Link in a Router, rather than mock a Router or pass come mocked context.
      */
      <StaticRouter>
        <InlineLink href="/news/articles/c0000000027o">
          Internal SPA article
        </InlineLink>
      </StaticRouter>,
    );
  });

  describe('with link not matching SPA route', () => {
    shouldMatchSnapshot(
      'should render correctly',
      <InlineLink href="https://www.bbc.com/news">BBC News</InlineLink>,
    );
  });
});
