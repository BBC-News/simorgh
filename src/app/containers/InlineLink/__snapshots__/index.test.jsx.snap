// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InlineLinkContainer external link accessibility should be explicitly marked "external" for screen reader users & localised 1`] = `
.c0 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
}

<a
  className="InlineLink-hkwpaz-0 guatGt"
  href="https://www.example.com/"
>
  این لینک هست
  <span
    className="c0"
  >
     ،لینک خارجی
  </span>
</a>
`;

exports[`InlineLinkContainer external link accessibility should be explicitly marked "external" for screen reader users 1`] = `
.c0 {
  -webkit-clip-path: inset(100%);
  clip-path: inset(100%);
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  height: 1px;
  overflow: hidden;
  position: absolute;
  width: 1px;
}

<a
  className="InlineLink-hkwpaz-0 guatGt"
  href="https://www.example.com/"
>
  This is a link
  <span
    className="c0"
  >
    , external
  </span>
</a>
`;

exports[`InlineLinkContainer internal link not matching SPA route should render correctly 1`] = `
<a
  className="InlineLink-hkwpaz-0 guatGt"
  href="https://www.bbc.com/news"
>
  <b>
    This is bold text for a link
  </b>
</a>
`;

exports[`InlineLinkContainer link matching routes for SPA should render correctly 1`] = `
<a
  className="InlineLink-hkwpaz-0-Link LLDYR"
  href="/news/articles/c0g992jmmkko"
  onClick={[Function]}
>
  This is text for an internal link
</a>
`;

exports[`InlineLinkContainer link matching routes for SPA should render correctly for TEST environment 1`] = `
<a
  className="InlineLink-hkwpaz-0-Link LLDYR"
  href="/news/articles/c0g992jmmkko"
  onClick={[Function]}
>
  This is text for an internal link
</a>
`;
